deployment/script/prerequisite.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+# This script implements the prerequisites for deploying CCNP, including installing docker, helm, python3-pip, 
+# and setting the access permissions of the TD device node and the ccnp working directory on the TD node.
+
+set -e
+
+INSTALL_DOCKER=true
+INSTALL_HELM=true
+INSTALL_PIP=true
+CCNP_PREREQS=true
+
+UDEV_FILE=/etc/udev/rules.d
+TDX_RULES_FILE=${UDEV_FILE}/90-tdx.rules
+CCNP_CONF=/usr/lib/tmpfiles.d/ccnp.conf
+
+function check_env {
+    if command -v docker &> /dev/null; then
+        INSTALL_DOCKER=false
+        echo "Skip: Docker has been installed."       
+    fi
+
+    if  command -v helm &> /dev/null; then
+        INSTALL_HELM=false
+        echo "Skip: Helm has been installed."
+    fi
+
+    if  command -v pip &> /dev/null; then
+        INSTALL_PIP=false
+        echo "Skip: Python3-pip has been installed."
+    fi
+}
+
+function install_docker {
+    #install GPG key
+    install -m 0755 -d /etc/apt/keyrings
+    rm -f /etc/apt/keyrings/docker.gpg
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+    chmod a+r /etc/apt/keyrings/docker.gpg
+
+    #install repo
+    echo \
+    "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+    "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+    tee /etc/apt/sources.list.d/docker.list > /dev/null
+    apt-get update > /dev/null
+
+    #install docker
+    apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+
+    #start docker
+    systemctl enable docker
+    systemctl daemon-reload
+    systemctl start docker
+}
+
+function install_helm {
+    #install repo
+    curl -fsSL https://baltocdn.com/helm/signing.asc | gpg --dearmor | tee /usr/share/keyrings/helm.gpg > /dev/null
+    echo \
+    "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/helm.gpg] https://baltocdn.com/helm/stable/debian/ all main" | \
+    tee /etc/apt/sources.list.d/helm-stable-debian.list > /dev/null
+    apt-get update > /dev/null
+
+    #install helm
+    apt-get install -y helm
+}
+
+function install_pip {
+    #install python3-pip
+    apt install -y python3-pip 
+}
+
+function ccnp_udev_rules { 
```

Comment:
You may want to check whether the files exist before running this function.

Reply:
Fixed. Thanks